src/main/java/net/sf/jabref/gui/PreviewPanel.java

```diff
@@ -275,16 +275,27 @@ public void setLayout(Layout layout) {
     }
 
     public void setEntry(BibEntry newEntry) {
-        if(entry.isPresent() && (entry.get() != newEntry)) {
-            entry.ifPresent(e -> e.removePropertyChangeListener(this));
-            newEntry.addPropertyChangeListener(this);
+
+        if (entry.isPresent() && (entry.get() != newEntry)) {
+            entry.ifPresent(e -> e.unregisterListener(this.currentChangeFieldUpdateEvent));
+            this.currentChangeFieldUpdateEvent = new ChangeFieldUpdateEvent();
+            newEntry.registerListener(this.currentChangeFieldUpdateEvent);
         }
+
         entry = Optional.ofNullable(newEntry);
 
         updateLayout();
         update();
     }
 
+    private class ChangeFieldUpdateEvent {
```

Comment:
No it doesnt. There I simply would write 
````java
entry.ifPresent(e -> e.unregisterListener(this));
newEntry.registerListener(this);
````
The same comment applies also to the EntryEditor.